apps/dashboard/src/components/workflow-editor/workflow-editor-provider.tsx

```diff
@@ -153,7 +162,27 @@ export const WorkflowEditorProvider = ({ children }: { children: ReactNode }) =>
     },
   });
 
-  const blocker = useBlocker(isDirty || isPending);
+  /*
+   * If there was a pending navigation when saving was in progress,
+   * proceed with that navigation now that changes are saved
+   *
+   * small timeout to briefly show the success dialog before navigating
+   */
+  const handleBlockedNavigation = useCallback(async () => {
+    return new Promise<void>((resolve) => {
+      setTimeout(() => {
+        blocker.proceed?.();
+        toast.dismiss();
+        resolve();
+      }, 500);
+    });
+  }, [blocker]);
+
+  const handleCancelNavigation = useCallback(() => {
+    if (blocker.state === 'blocked') {
+      blocker.reset();
+    }
+  }, [blocker]);
```

Comment:
fallback in case the user would be stuck on the "Saving changes" dialog (API timeout etc). Now he can press the "X" button otherwise he would need to refresh to get out of that state.